modern-publish-plugin.md

```diff
@@ -0,0 +1,211 @@
+or removed.
+
+### Plugins
+
+Specific plugins for each supported project type:
+- `com.vanniktech.maven.publish.java`
+- `com.vanniktech.maven.publish.gradle`
+- `com.vanniktech.maven.publish.android`
+- `com.vanniktech.maven.publish.kotlin.js`
+- `com.vanniktech.maven.publish.kotlin.mpp`
+
+There are two reasons for going with individual plugins. 
+
+
+On top of that there will be a `com.vanniktech.maven.publish.base` plugin that 
+gets applied by all of these and the existing `com.vanniktech.maven.publish` as 
+described above.
+
+#### `com.vanniktech.maven.publish.base`
+
+- applies `maven-publish` Gradle plugin
+- provides the API that is shared for all project types
+- can be used for shared configurations using `plugins.withId(...)` in projects with mixed types
+
+```
+mavenPublish {
+  # applies signing plugin and tells Gradle to sign all publications
+  signAllPublications()
+  # adds the Sonatype release/snapshot repo as targets based on version
+  # uses Gradle credentials API for username/password
+  # automatically calls signAllPublications()
+  # sets nexus.baseUrl
+  publishToMavenCentral()
+
+  # might not be needed as it defaults to the mutable `project.getGroup()`
+  groupId = "com.example"
+  # might not be needed as it defaults to the mutable `project.getGroup()`
+  version = "1.0.0"
+  # Base because their might be multiple artifacts produced by one module.
+  # Without this the arifact id would always be based on `project.getName()`.
+  # Implementation in Kotlin MPP is kind of hacky using replace so it might be
+  # best to just not support it. How do we handle compatibility?
+  baseArtifactId = "library"
+
+  # Same API as Gradle's, can probably just use their interfaces/classes
+  # Will be applied to all publications
+  pom {
+    name = "My Library"
+    description = "A description of what my library does."
+    inceptionYear = "2020"
+    url = "https://github.com/username/mylibrary/"
+    
+    scm {
+        connection = "scm:git:git://github.com/username/mylibrary.git"
+        developerConnection = "scm:git:ssh://git@github.com/username/mylibrary.git"
+        url = "https://github.com/username/mylibrary/"
+    } 
+    
+    licenses {
+        license {
+            name = "The Apache Software License, Version 2.0"
+            url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+            distribution = "repo"
+        }
+    }
+    developers {
+        developer {
+            id = "username"
+            name = "User Name"
+            url = "https://github.com/username/"
+        }
+    }
+  }
+
+  # Configure closeAndReleaseRepository
+  nexus {
+     baseUrl = "https://your_nexus_instance"
+     stagingProfile = "net.example"
+     # Start supporting the Gradle credentials API
+     respositoryUserName = "username"
+     respositoryPassword = "password"
+  }
+}
+```
+
+#### `com.vanniktech.maven.publish.java`
+
+Meant to be used with `java`, `java-library`, `org.jetbrains.kotlin.jvm`. 
```

Comment:
Should there be a `com.vanniktech.maven.publish.kotlin.jvm`  that's just an alias to `com.vanniktech.maven.publish.java`? For consistency with the other Kotlin plugins.

Reply:
Yup